app/models/mixins/vim_connect_mixin.rb

```diff
@@ -17,15 +20,11 @@ def connect(options = {})
     # The following require pulls in both MiqFaultTolerantVim and MiqVim
     require 'VMwareWebService/miq_fault_tolerant_vim'
 
-    if options[:fault_tolerant]
-      options[:ems] = self
-      MiqFaultTolerantVim.new(options)
-    else
-      ip   = options[:ip] || hostname
-      user = options[:user] || authentication_userid(options[:auth_type])
-      pass = options[:pass] || authentication_password(options[:auth_type])
-      MiqVim.new(ip, user, pass)
-    end
+    options[:ip] ||= hostname
+    options[:user] ||= authentication_userid(options[:auth_type])
+    options[:pass] ||= authentication_password(options[:auth_type])
+
+    self.class.make_connection(options)
```

Comment:
So lets leave this the way it currently is, we'll have two similar code paths for connect but when you do your followup to split `raw_connect` and `validate` we can move this over to `raw_connect`